icu4j/main/core/src/main/java/com/ibm/icu/text/NFSubstitution.java

```diff
@@ -1225,7 +1225,11 @@ class FractionalPartSubstitution extends NFSubstitution {
         } else {
             byDigits = false;
             useSpaces = true;
-            this.ruleSet.makeIntoFractionRuleSet();
+            if (this.ruleSet != null) {
+                this.ruleSet.makeIntoFractionRuleSet();
+            } else {
+                throw new IllegalArgumentException("rule set is missing");
```

Comment:
Please use a parse exception instead of illegal argument exception to mirror the error in ICU4C.